numba/parfors/array_analysis.py

```diff
@@ -1758,6 +1757,13 @@ def update_replacement_slice(
             known,
         )
 
+    def _get_unique_var_name(self, prefix):
+        idx = 0
+        for var_name in self.typemap.keys():
+            if var_name.startswith(prefix):
+                idx += 1
+        return f"{prefix}_{idx}"
```

Comment:
Yes, sort of. But `mk_unique_var` depends on a global state, whereas the current approach uses a local state on ArrayAnalysis class.

Reply:
What problem was the global creating?  Incrementing a global seems a lot more efficient that iterating through the entire typemap.  I think you do need something like this because if a variable is going to be used in multiple scopes then you will run into this uniqueness issue if all you look at is the local scope where you first create the variable.